src/VirtoCommerce.ExperienceApiModule.Core/Services/DynamicPropertyResolverService.cs

```diff
@@ -35,7 +35,7 @@ public async Task<IEnumerable<DynamicPropertyObjectValue>> LoadDynamicPropertyVa
             criteria.Take = int.MaxValue;
             var searchResult = await _dynamicPropertySearchService.SearchDynamicPropertiesAsync(criteria);
 
-            var existingDynamicProperties = searchResult.Results.Where(p => entity.DynamicProperties.All(x => x.Id == p.Id || x.Name.EqualsInvariant(p.Name)));
+            var existingDynamicProperties = searchResult.Results.Where(p => (entity.DynamicProperties ?? new DynamicObjectProperty[] { }).All(x => x.Id == p.Id || x.Name.EqualsInvariant(p.Name)));
```

Comment:
It`s ok to move predicate to local/private function.
```suggestion
            var existingDynamicProperties = searchResult.Results
                .Where(p => entity.DynamicProperties != null 
                         && entity.DynamicProperties.All(x => x.Id == p.Id 
                                                      || x.Name.EqualsInvariant(p.Name)));
```